File: 1975-maximum-matrix-sum/1975-maximum-matrix-sum.cpp
class Solution {
public:
    long long maxMatrixSum(vector<vector<int>>& matrix) {
        long long sum = 0;
        int smallestAbs = INT_MAX;
        bool oddNeg = false;
        bool containsZero = false;
        for(int i = 0; i < matrix.size(); i++)
        {
            for(int j = 0; j < matrix.size(); j++)
            {
                sum += abs(matrix[i][j]);
                smallestAbs = min(smallestAbs, abs(matrix[i][j]));
                if(matrix[i][j] < 0)
                {
                    oddNeg = !oddNeg;
                }
                else if(matrix[i][j] == 0)
                {
                    containsZero = true;
                }
            }
        }
        if(oddNeg && !containsZero)
        {
            return sum - (2 * smallestAbs);
        }
        return sum;
    }
};

File: 1975-maximum-matrix-sum/1975-maximum-matrix-sum.py
class Solution:
    def maxMatrixSum(self, matrix: List[List[int]]) -> int:
        tot = 0
        minNum = float('inf')
        containsZero = False
        oddNeg = False
        for row in matrix:
            for cell in row:
                if cell == 0:
                    containsZero = True
                else:
                    if cell < 0:
                        oddNeg = not oddNeg
                    minNum = min(minNum, abs(cell))
                tot += abs(cell)
        if oddNeg and  not containsZero:
            return tot - 2 * minNum
        return tot